app/src/main/java/ayds/lisboa/songinfo/moredetails/fulllogic/OtherInfoWindow.kt

```diff
@@ -0,0 +1,204 @@
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_other_info)
+        initTextPaneArtistInfo()
+        initDataBase()
+        initLastFMAPI()
+        initImageView()
+        initOpenURLButton()
+        getArtistInfo(intent.getStringExtra(ARTIST_NAME)?:"")
+    }
+
+    private fun initTextPaneArtistInfo(){
+        textPaneArtistBio = findViewById(R.id.textPane2)
+    }
+
+    private fun initDataBase(){
+        dataBase = DataBase(this)
+    }
+
+    private fun initLastFMAPI(){
+        lastFMAPI = createRetrofit().create(LastFMAPI::class.java)
+    }
+
+    private fun createRetrofit() =
+        Retrofit.Builder()
+            .baseUrl(RETROFIT_BASE_URL)
+            .addConverterFactory(ScalarsConverterFactory.create())
+            .build()
+
+    private fun initImageView() {
+        imageView = findViewById<View>(R.id.imageView) as ImageView
+    }
+
+    private fun initOpenURLButton() {
+        openUrlButton = findViewById<View>(R.id.openUrlButton) as Button
+    }
+
+    private fun getArtistInfo(artistName: String) {
+        initArtistName(artistName)
+        createThreadToGetArtistInfo()
+    }
+
+    private fun initArtistName(artistName: String){
+        this.artistName = artistName
+    }
+
+    private fun createThreadToGetArtistInfo() {
+        Thread {
+            setArtistInfoInView(getArtistInfoFromDataBaseOrService())
+        }.start()
+    }
+
+    private fun setArtistInfoInView(artistInfo: String) {
+        setExternalServiceImg()
+        setArtistBioInTextPane(artistInfo)
+    }
+
+    private fun setExternalServiceImg() {
+        runOnUiThread {
+            Picasso.get().load(URL_IMAGE).into(imageView)
+        }
+    }
+
+    private fun setArtistBioInTextPane(artistInfo: String){
+        runOnUiThread {
+            textPaneArtistBio.text = Html.fromHtml(artistInfo)
+        }
+    }
+
+    private fun getArtistInfoFromDataBaseOrService() : String {
+        var artistInfo = getArtistInfoFromDataBase()
+        if (existInDataBase(artistInfo)) {
+            artistInfo = "$LOCAL_DATABASE_PREFIX$artistInfo"
+        } else {
+            artistInfo = getArtistInfoFromService()
+            saveArtistInDataBase(artistInfo)
+        }
+        return artistInfo
+    }
+
+    private fun getArtistInfoFromDataBase() =
+        dataBase.getInfo(artistName)
+
+    private fun existInDataBase(artistInfo: String?) =
+        artistInfo != null
+
+    private fun getArtistInfoFromService(): String {
+        lateinit var queryArtistInfo: JsonObject
+        lateinit var artistBiography: JsonElement
+        try {
+            queryArtistInfo = getQueryBodyOfArtistInfoFromService()
+            artistBiography = getArtistBiography(queryArtistInfo)
+            setOnClickerListenerToOpenURLButton(queryArtistInfo)
+        } catch (e1: IOException) {
+            e1.printStackTrace()
+        }
+        return getStringArtistInfoFromService(artistBiography)
+    }
```

Comment:
Esto es algo peligroso. Si hay una excepción se hace el catch, y luego se intenta acceder al late init, crash en ejecución. Mejor:
```suggestion
    private fun getArtistInfoFromService(): String {
        lateinit var queryArtistInfo: JsonObject
        lateinit var artistBiography: JsonElement
        return try {
            val queryArtistInfo = getQueryBodyOfArtistInfoFromService()
            val artistBiography = getArtistBiography(queryArtistInfo)
            setOnClickerListenerToOpenURLButton(queryArtistInfo)
            getStringArtistInfoFromService(artistBiography)
        } catch (e1: IOException) {
            e1.printStackTrace()
            ""
        }
    }
```